src/layercode.js

```diff
@@ -0,0 +1,317 @@
+                    }
+                    if(self.layer == "odorreport"){
+                        zoom = self._map.getZoom(), origin = self._map.getCenter() ;
+                        Layer_URL = "https://odorlog.api.ushahidi.io/api/v3/posts/" ;
+                    }
+                    if(self.layer == "mapknitter"){
+                        zoom = self._map.getZoom(), northeast = self._map.getBounds().getNorthEast() , southwest = self._map.getBounds().getSouthWest() ;
+                        Layer_URL = "https://mapknitter.org/map/region/Gulf-Coast.json?minlon="+(southwest.lng)+"&minlat="+(southwest.lat)+"&maxlon="+(northeast.lng)+"&maxlat="+(northeast.lat);
+                    }
+
+                    if(typeof self._map.spin === 'function'){
+                        self._map.spin(true) ;
+                    }
+                    $.getJSON(Layer_URL , function(data){
+                    if(self.layer == "fractracker")
+                        self.parseData(data.feed.entry);
+                    else
+                        self.parseData(data) ;
+                    if(typeof self._map.spin === 'function'){
+                        self._map.spin(false) ;
+                    }
+                    });
+                })();
+
+
+        },
+
+        getMarker: function(data) {
+            if(this.layer == "fractracker"){
+                var redDotIcon = new L.icon.fracTrackerIcon();
+                var props = ["timestamp", "name", "summary", "website", "contact", "email", "phone", "streetaddress", "city", "state", "zipcode", "latitude", "longitude", "category"];
+                var item = {};
+                props.forEach(function(element) {
+                    item[element] = data["gsx$" + element]["$t"];
+                });
+
+                item["updated"] = data.updated.$t;
+                item["use"] = (data.gsx$useformap.$t.replace(/\s+/g, '').toLowerCase() === "use");
+                item["latitude"] = item["latitude"].replace(/[^\d.-]/g, "");
+                item["latitude"] = item["latitude"].replace(/[^\d.-]/g, "");
+
+                var fracTracker;
+                fracTracker = L.marker([item["latitude"], item["longitude"]], {
+                    icon: redDotIcon
+                }).bindPopup(this.generatePopup(item));
+
+                return fracTracker;
+           }
+
+           if(this.layer == "skytruth"){
+                var redDotIcon =new L.icon.skyTruthIcon();
+                var lat = data.lat ;
+                var lng = data.lng;
+                var title = data.title ;
+                var url = data.link ;
+                var skymarker ;
+                if (!isNaN(lat) && !isNaN(lng) ){
+                  skymarker = L.marker([lat , lng] , {icon: redDotIcon}).bindPopup("<a href="+url+">" +title + "</a><br>" + "<br><strong> lat: " + lat + "</strong><br><strong> lon: " + lng + "</strong> <br><br>Data provided by <a href='http://alerts.skytruth.org/'>alerts.skytruth.org/</a>") ;
```

Comment:
Hi...can we structure this as we have done for mapknitter in line 136 for easy code readability .